### PR TITLE
Circleci project setup

### PR DESCRIPTION
**Pull request by rulixxx**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          name: Build Docker image
+          command: |
+            set -e
+            BRANCH_OR_TAG="${CIRCLE_TAG:-$CIRCLE_BRANCH}"
+            CLEAN_BRANCH=$(echo $BRANCH_OR_TAG | tr / -)
+            echo "$DOCKERHUB_PASSWORD" | docker login -u "$DOCKERHUB_USER" --password-stdin
+            echo "$QUAYIO_PASSWORD" | docker login -u "$QUAYIO_USER" --password-stdin quay.io
+            docker build -t $IMAGE_NAME:$CLEAN_BRANCH .
+            echo 'Verify program(s) from this repo'
+            docker run -t --rm $IMAGE_NAME:$CLEAN_BRANCH bash -c "java com.hartwig.hmftools.gripss.GripssApplicationKt  | grep 'Missing required options' && echo 'gripss OK'"
```
Comment (keiranmraine) [inline .circleci/config.yml:33]: Does `GripssApplicationKt` not have a version number we can test for that doesn't give a non-zero exit code?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    docker:
+      - image: circleci/buildpack-deps:stretch
+    steps: 
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Build Docker image
+          command: |
+            set -e
+            BRANCH_OR_TAG="${CIRCLE_TAG:-$CIRCLE_BRANCH}"
```
Comment (keiranmraine) [inline .circleci/config.yml:26]: The quay.io push fails as you need to do the docker logins like cgpPindel does:

https://github.com/cancerit/cgpPindel/blob/40b1e8a7e4a7774660c32513673fecd538d51a0b/.circleci/config.yml#L22-L23

**Review by keiranmraine — CHANGES_REQUESTED** [review]
Changes needed, see conversations.  You can re-request review on the existing pull via the UI.

File: .circleci/config.yml (hunk 1 above)
Comment (rulixxx) [reply .circleci/config.yml:33]: No there is no version nor help option. Plus it throws a 1 exit status with no arguments or wrong arguments. However using this grep trick seems to produce the desired behaviour.

**Comment by rulixxx** [comment]
Made all suggested changes from review

**Pull request merged**